src/keyboard.c

```diff
@@ -303,14 +315,29 @@ handle_compositor_keybindings(struct keyboard *keyboard,
 
 	/* Handle compositor key bindings */
```

Comment:
Suggest expanding the comment here to explain why. Maybe something like this:

```
	/*
	 * Handle compositor keybinds
	 * 
	 * When matching against keybinds, we process the input keys in the
	 * following order of precedence:
	 *   a. Keycodes (of physical keys)
	 *   b. Translated keysyms (taking into account modifiers, so if Shift+1
	 *      were pressed on a us keyboard, the keysym would be !)
	 *   c. Raw keysyms (ignoring modifiers such as shift, so in the above
	 *      example the keysym would just be 1)
	 *
	 * The reasons for this approach are:
	 *   1. To make keybinds keyboard-layout agnostic (by checking keycodes
	 *      before keysyms). This means that in a multi-layout situation,
	 *      keybinds work regardless of which layout is active at the time
	 *      of the key-press.
	 *   2. To support keybinds relating to keysyms that are only available
	 *      in a particular layout, for example å, ä and ö
	 *   3. To differentiate between keybinds `S-ö` and `S-Ö` (by checking
	 *      translated before raw)
	 *   4. To support keybinds such as `S-1` (by checking raw keysyms)
	 */
```

Reply:
Slightly modified and updated.